cwf/k8s/cwf_operator/pkg/controller/hacluster/hacluster_controller.go

```diff
@@ -147,47 +152,73 @@ func (r *ReconcileHACluster) Reconcile(request reconcile.Request) (reconcile.Res
 		reqLogger.Error(err, "An error occurring while fetching standby health status")
 	} else {
+		hacluster.Status.ConsecutiveStandbyErrors = 0
 		reqLogger.Info("Fetched standby health status", "health", standbyHealth.Health.String(), "message", standbyHealth.HealthMessage)
 	}
 
 	failover := false
 	var initErr error
 	var updatedStatus magmav1alpha1.HAClusterStatus
 	if activeHealth.Health == protos.HealthStatus_HEALTHY {
-		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status)
+		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status, false)
 	} else if activeHealth.Health == protos.HealthStatus_UNHEALTHY && standbyHealth.Health == protos.HealthStatus_HEALTHY {
-		failover = true
 		reqLogger.Info("Promoting standby due to unhealthy active", "promoted active", standbyGateway)
 		hacluster.Status.ActiveInitState = magmav1alpha1.Uninitialized
 		hacluster.Status.StandbyInitState = magmav1alpha1.Uninitialized
-		updatedStatus, initErr = r.initCluster(standbyGateway, activeGateway, request.Namespace, hacluster.Status)
+		hacluster.Status.ConsecutiveActiveErrors = 0
+		hacluster.Status.ConsecutiveStandbyErrors = 0
+		updatedStatus, initErr = r.initCluster(standbyGateway, activeGateway, request.Namespace, hacluster.Status, true)
 	} else {
 		reqLogger.Info("Both gateways are detected to be unhealthy")
-		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status)
+		// If both gateways are unhealthy, there is a chance that a restart
+		// occurred, rendering the gateway(s) uninitialized. As a precaution,
+		// update the status to uninitialized and re-init. If the gateways are
+		// already initialized, this will be a no-op
+		hacluster.Status.ActiveInitState = magmav1alpha1.Uninitialized
+		hacluster.Status.StandbyInitState = magmav1alpha1.Uninitialized
+		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status, false)
 	}
 
 	if initErr != nil {
-		reqLogger.Error(initErr, "failover occurred", strconv.FormatBool(failover))
+		reqLogger.Error(initErr, "did failover occur", strconv.FormatBool(failover))
 	}
 	hacluster.Status = updatedStatus
 	updateErr := r.client.Status().Update(context.TODO(), hacluster)
 	if updateErr != nil && failover {
-		err = fmt.Errorf("Updating hacluster status with promoted active %s failed; %s", standbyGateway, updateErr)
-		return reconcile.Result{RequeueAfter: r.reconcilePeriod}, updateErr
+		reqLogger.Error(updateErr, "Updating hacluster status with promoted active failed", "promoted active", standbyGateway)
+		return reconcile.Result{}, updateErr
 	} else if updateErr != nil {
-		return reconcile.Result{RequeueAfter: r.reconcilePeriod}, updateErr
+		reqLogger.Error(updateErr, "Updating hacluster status failed")
+		return reconcile.Result{}, updateErr
 	}
```

Comment:
I think this will read better if you nest `failover` conditional inside the `updateErr` nilcheck conditional